File: modules/mex/examples/mex_engine_demo_2.c
//=============================================================================
// Copyright (c) 2016-present Allan CORNET (Nelson)
//=============================================================================
// This file is part of Nelson.
//=============================================================================
// LICENCE_BLOCK_BEGIN
// SPDX-License-Identifier: LGPL-3.0-or-later
// LICENCE_BLOCK_END
//=============================================================================
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include "engine.h"
//=============================================================================
int
main(int argc, char* argv[])
{
    Engine* ep = NULL;
    char buffer[256], cmd[256];
    if (!(ep = engOpen(""))) {
        fprintf(stderr, "\nCan't start Nelson engine.\n");
        exit(-1);
    }

    engOutputBuffer(ep, buffer, 256);

    while (strcmp(cmd, "exit\n") != 0) {
        printf("\nEnter a Nelson command to evaluate.\n");
        printf("For example: X = 1:5\n");
        printf("To finish: exit\n");
        printf(">> ");
        fgets(cmd, 255, stdin);

        engEvalString(ep, cmd);
        printf("%s", buffer);
    }
    engClose(ep);
    return EXIT_SUCCESS;
}
//=============================================================================

File: modules/mex/examples/mex_engine_demo_2.m
%=============================================================================
% Copyright (c) 2016-present Allan CORNET (Nelson)
%=============================================================================
% This file is part of Nelson.
%=============================================================================
% LICENCE_BLOCK_BEGIN
% SPDX-License-Identifier: LGPL-3.0-or-later
% LICENCE_BLOCK_END
%=============================================================================
destinationdir = [tempdir(), 'mex_engine_demo_2/'];
if isdir(destinationdir)
  rmdir(destinationdir, 's');
end
mkdir(destinationdir);
destination = [destinationdir, 'mex_engine_demo_2.c'];
status = copyfile('mex_engine_demo_2.c', destinationdir);
cd(destinationdir);
mex('-client', 'engine', 'mex_engine_demo_2.c');
generated_executable = [destinationdir, 'mex_engine_demo_2'];
disp('Open an Terminal')
disp('Set environment variables to find Nelson (see doc mex).')
disp('Go to:')
disp(destinationdir)
disp('Launch')
disp('mex_engine_demo_2')
%=============================================================================
